src/librustc_mir/interpret/intrinsics.rs

```diff
@@ -37,6 +38,49 @@ fn numeric_intrinsic<'tcx, Tag>(
     Ok(Scalar::from_uint(bits_out, size))
 }
 
+
+crate fn eval_nulary_intrinsic<'tcx>(
```

Comment:
Could use a comment re. why this construct exists (+ what it does but that is overall less important).